### PR TITLE
Isy 919 code refactoring

### PR DESCRIPTION
**Pull request by msg-ptavoul**
Branch includes:
1. ISY-722: Replacing structural directives with Angular 17 control flow
2. Typo fixes
3. Test fixes
4. Code Refactoring
5. Default language changed
6. Updated dependencies

### CODE REVIEW

File: projects/isy-angular-widgets/src/lib/hauptfenster/hauptfenster.component.html [hunk 1]
```diff
@@ -42,73 +45,81 @@
...
+    @if (showLinksnavigation) {
+      <div
+        class="isy-hauptfenster-linksnavigation dont-print"
+        [style.width]="collapsedLinksnavigation ? null : linksNavigationWidth"
+      >
+        <div [hidden]="collapsedLinksnavigation" id="open-links-navigation">
+          @if (allowSidebarCollapse) {
+            <div class="collapseButton">
+              <button
+                *ngIf="allowSidebarCollapse"
```
Comment (mtpham-collab) [inline projects/isy-angular-widgets/src/lib/hauptfenster/hauptfenster.component.html:67]: The if instruction inside the button no longer seems to be necessary

**Review by mtpham-collab — CHANGES_REQUESTED** [review]

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 2]
```diff
@@ -133,264 +136,277 @@
...
             </span>
           </div>
         </div>
-        <div class="col-12">
+        <div class="col-12 xl:col-6">
           <div class="field">
             <span class="p-float-label">
               <p-calendar
-                id="geburtdatum"
+                id="edit-geburtsdatum"
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:190]: Done

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 3]
```diff
@@ -133,264 +136,277 @@
...
+              <span class="p-float-label">
+                <input
+                  id="person-id"
+                  type="text"
+                  class="w-full"
+                  pInputText
+                  [formControlName]="'id'"
+                  (focus)="onFormControlFocus(idForm.controls.id)"
+                />
+                <label id="id-label" for="person-id">ID <demo-required-label></demo-required-label></label>
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:307]: Done

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 4]
```diff
@@ -133,264 +136,277 @@
...
+              <span class="p-float-label">
+                <input
+                  id="geburtsort"
+                  class="w-full"
+                  type="text"
+                  pInputText
+                  [formControlName]="'geburtsort'"
+                  (focus)="onFormControlFocus(geburtsInformationenForm.controls.geburtsort)"
+                />
+                <label id="geburtsort-label" class="align-content-end" for="geburtsort">
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:355]: Unused ID removed

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 5]
```diff
@@ -133,264 +136,277 @@
...
+              <span class="p-float-label">
+                <input
+                  id="geburtsname"
+                  class="w-full"
+                  type="text"
+                  pInputText
+                  [formControlName]="'geburtsname'"
+                  (focus)="onFormControlFocus(geburtsInformationenForm.controls.geburtsname)"
+                />
+                <label id="geburtsname-label" class="align-content-end" for="geburtsname">
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:337]: Unused ID removed

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 6]
```diff
@@ -133,264 +136,277 @@
...
+                <p-calendar
+                  id="geburtdatum"
+                  dateFormat="dd.mm.yy"
+                  [formControlName]="'geburtsdatum'"
+                  [showIcon]="true"
+                  appendTo="body"
+                  (focusout)="onFormControlFocus(geburtsInformationenForm.controls.geburtsdatum)"
+                >
+                </p-calendar>
+                <label id="geburtsdatum-label" class="align-content-end" for="geburtdatum">
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:374]: Unused ID removed

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 7]
```diff
@@ -133,264 +136,277 @@
...
+                  <demo-required-label></demo-required-label>
+                </label>
+              </span>
+            </div>
+          </div>
+          <div class="col-12">
+            <div class="field">
+              <span class="p-float-label">
+                <p-calendar
+                  id="geburtdatum"
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:366]: Done

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 8]
```diff
@@ -133,264 +136,277 @@
...
+                  <demo-required-label></demo-required-label>
+                </label>
+              </span>
+            </div>
+          </div>
+          <div class="col-12">
+            <div class="field">
+              <span class="p-float-label">
+                <p-dropdown
+                  id="staatangehoerigkeit"
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:385]: Done

File: projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html [hunk 9]
```diff
@@ -133,264 +136,277 @@
...
+                  appendTo="body"
+                  (focusout)="onFormControlFocus(geburtsInformationenForm.controls.staatsangehoerigkeit)"
+                >
+                  <ng-template let-land pTemplate="item">
+                    <div class="country-item">
+                      <span>{{ land.name }}</span>
+                    </div>
+                  </ng-template>
+                </p-dropdown>
+                <label id="staatsangehoerigkeit-label" class="align-content-end" for="staatangehoerigkeit">
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets-demo/src/app/feature/objekt-suchen/objekt-suchen.component.html:401]: Unused ID removed

File: projects/isy-angular-widgets/src/lib/hauptfenster/hauptfenster.component.html (hunk 1 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/hauptfenster/hauptfenster.component.html:67]: Removed unnecessary *ngif

File: projects/isy-angular-widgets/src/lib/input-char/components/input-char-dialog/input-char-dialog.component.html [hunk 10]
```diff
@@ -41,8 +41,9 @@
     >
     </p-selectButton>
     <div id="lower-right-panel">
-      <isy-input-char-preview *ngIf="selectedZeichenObjekt" [zeichenObjekt]="selectedZeichenObjekt" class="flex-grow-1">
-      </isy-input-char-preview>
+      @if (selectedZeichenObjekt) {
+        <isy-input-char-preview [zeichenObjekt]="selectedZeichenObjekt" class="flex-grow-1"> </isy-input-char-preview>
```
[the remark this answers is not in the document]
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/input-char/components/input-char-dialog/input-char-dialog.component.html:45]: Space removed

**Pull request merged**